content/docs/2.12/concepts/scaling-deployments.md

```diff
@@ -201,6 +201,80 @@ Starting from Kubernetes v1.18 the autoscaling API allows scaling behavior to be
 
 **Assumptions:** KEDA must be running on Kubernetes cluster v1.18+, in order to be able to benefit from this setting.
 
+---
+
+```yaml
+advanced:
+  complexScalingLogic:                                    # Optional. Section to specify complex scaling logic
+    target: {target-value-to-scale-on}                    # Mandatory. New target if metrics are anyhow composed together
+    formula: {formula-for-fetched-metrics}                # Mandatory. Formula for calculation
+```
+
+**`complexScalingLogic:`**
+
+The `complexScalingLogic` is optional. If defined, both `target` and `formula` are mandatory. Using this structure creates `composite-scaler-metric` for the HPA that will replace all requests for external metrics and handle then internally. With `complexScalingLogic` all external triggers' names **must** be defined, for potentially being used in the `formula` itself.
```

Comment:
I presume we have validation on the trigger names?


Same here, end-users don't have to care about HPA. Also, I'm removing external as KEDA end-users might/will be confused to think this is only supported with external scalers and not built-in ones.
```suggestion
The `complexScalingLogic` is optional. If defined, both `target` and `formula` are mandatory. By using formula-based scaling logic, KEDA will compute the metric to make autoscaling decisions on. With `complexScalingLogic` all triggers' names **must** be defined, for potentially being used in the `formula` itself.
```

Reply:
there is validation for having the trigger names defined when `formula` is specified in yaml. Currently oyu have to specify all triggers because there is no parsing. Iam currently working on parsing the formula to extract the names and match them to trigger names so user can define trigger names to only those triggers that are used in the formula which would be more intuitive I think.

I will follow up here